Source/AudioVisualizer.cpp

```
@@ -615,62 +706,151 @@ void AudioVisualizer::renderCompactMeter(NUIRenderer& renderer) {
+        NUIRect clipRect(leftMeter.x, leftMeter.y - 1.0f, leftMeter.width, 3.0f);
+        renderer.fillRoundedRect(clipRect, 1.0f, NUIColor(1.0f, 0.15f, 0.15f, leftClipIndicator_));
+    }
+    if (rightClipIndicator_ > 0.02f) {
+        NUIRect clipRect(rightMeter.x, rightMeter.y - 1.0f, rightMeter.width, 3.0f);
+        renderer.fillRoundedRect(clipRect, 1.0f, NUIColor(1.0f, 0.15f, 0.15f, rightClipIndicator_));
     }
 }
 
+void AudioVisualizer::renderCompactWaveform(NUIRenderer& renderer) {
+    NUIRect bounds = getBounds();
+    if (bounds.isEmpty()) return;
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)